File: app/src/main/java/com/lenar/healthyfit/screens/MainActivity.java
package com.lenar.healthyfit.screens;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.FrameLayout;

import com.lenar.healthyfit.R;
import com.lenar.healthyfit.fragments.AddNewEntriesFragment;
import com.lenar.healthyfit.listeners.OnCityClickListener;

import static com.lenar.healthyfit.screens.ConcreteCityActivity.KEY_CITY;

public class MainActivity extends AppCompatActivity implements OnCityClickListener {

    private FrameLayout fragment_container;
    private DrawerLayout drawer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        startActivity(new Intent(this, HistoryActivity.class));
        fragment_container = findViewById(R.id.fragment_container);
        FragmentManager myFragmentManager;
        myFragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = myFragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragment_container, new AddNewEntriesFragment());
        fragmentTransaction.commit();

        Toolbar toolbar = findViewById(R.id.tb_main);
        toolbar.setTitle("Добавить новые записи");
        setSupportActionBar(toolbar);

        drawer = findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close) {
            @Override
            public void onDrawerSlide(View drawerView, float slideOffset) {
                if (slideOffset != 0) {
                    // started opening
                    View view = getCurrentFocus();
                    if (view != null) {
                        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
                        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
                    }
                }
                super.onDrawerSlide(drawerView, slideOffset);
            }
        };
        drawer.addDrawerListener(toggle);
        toggle.syncState();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_settings) {
            Intent settingIntent = new Intent(this, SettingsActivity.class);
            startActivity(settingIntent);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onCityClick(int position) {
        Intent concreteCityIntent = new Intent(this, ConcreteCityActivity.class);
        concreteCityIntent.putExtra(KEY_CITY, position);
        startActivity(concreteCityIntent);
    }
}

File: app/src/main/java/com/nilden/fitforlife/screens/MainActivity.kt
package com.nilden.fitforlife.screens

import android.Manifest
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.v4.app.ActivityCompat
import android.support.v4.content.ContextCompat
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.inputmethod.InputMethodManager
import com.nilden.fitforlife.R
import com.nilden.fitforlife.fragments.AddNewEntriesFragment
import com.nilden.fitforlife.fragments.HistoryFragment
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.app_bar_main.*


class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, AddNewEntriesFragment()).commit()

        toolbar.setTitle(R.string.add_new_entries)
        setSupportActionBar(toolbar)

        val toggle = object : ActionBarDrawerToggle(this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close) {

            override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
                if (slideOffset == 0f) {
                    // drawer closed
                } else if (slideOffset != 0f) {
                    // started opening
                    val view = currentFocus
                    if (view != null) {
                        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                        imm.hideSoftInputFromWindow(view.windowToken, 0)
                    }
                }
                super.onDrawerSlide(drawerView, slideOffset)
            }
        }
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
        val navigationView = findViewById<NavigationView>(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(this)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_settings) {
            val settingIntent = Intent(this, SettingsActivity::class.java)
            startActivity(settingIntent)
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    private val REQUEST_WAKE_LOCK = 1012
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId

        if (id == R.id.add_new_entries) {
            // Handle the camera action
            toolbar.setTitle(R.string.add_new_entries)
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, AddNewEntriesFragment()).commit()
        } else if (id == R.id.history) {
            toolbar.setTitle(R.string.history_of_entries)
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, HistoryFragment()).commit()
        } else if (id == R.id.pedometer) {

            val permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.WAKE_LOCK)

            if (permissionCheck != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.WAKE_LOCK), REQUEST_WAKE_LOCK)
            } else {
                startPedometer()
            }
        } else if (id == R.id.settings) {
            val settingIntent = Intent(this, SettingsActivity::class.java)
            startActivity(settingIntent)
        } else if (id == R.id.nav_share) {

        } else if (id == R.id.nav_send) {

        }
        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        when (requestCode) {
            REQUEST_WAKE_LOCK -> if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                startPedometer()
            }

            else -> {
            }
        }
    }

    private fun startPedometer() {
        val settingIntent = Intent(this, PedometerActivity::class.java)
        startActivity(settingIntent)
    }
}
